WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEmailFragment.java

```diff
@@ -0,0 +1,328 @@
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.BuildConfig;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.network.rest.wpcom.account.AccountRestClient;
+import org.wordpress.android.fluxc.store.AccountStore.OnAvailabilityChecked;
+import org.wordpress.android.ui.accounts.login.nav.LoginNav;
+import org.wordpress.android.ui.accounts.login.nav.LoginStateGetter;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.EditTextUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.emailchecker2.EmailChecker;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+public class LoginEmailFragment extends Fragment implements TextWatcher {
+    private static final String KEY_IN_PROGRESS = "KEY_IN_PROGRESS";
+
+    public static final String TAG = "login_email_fragment_tag";
+    public static final int MAX_EMAIL_LENGTH = 100;
+
+    private TextInputLayout mEmailEditTextLayout;
+    private EditText mEmailEditText;
+    private Button mNextButton;
+    private ProgressDialog mProgressDialog;
+
+    private LoginNav.InputEmail mLoginNavInputEmail;
+    private boolean mEmailAutoCorrected;
+
+    @Inject Dispatcher mDispatcher;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
+
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.login_email_screen, container, false);
+
+        mEmailEditText = (EditText) rootView.findViewById(R.id.login_email);
+        mEmailEditText.addTextChangedListener(this);
+
+        mEmailEditTextLayout = (TextInputLayout) rootView.findViewById(R.id.login_email_layout);
+
+        mNextButton = (Button) rootView.findViewById(R.id.login_email_next_button);
+        mNextButton.setOnClickListener(new OnClickListener() {
+                public void onClick(View v) {
+                    next(getCleanedEmail());
+                }
+        });
+
+        autoFillFromBuildConfig();
+
+        mEmailEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (event != null
+                        && event.getAction() == KeyEvent.ACTION_UP
```

Comment:
I know this spacing is completely correct, but it makes me cringe for some reason 🌵